### PR TITLE
sysinfo.cc: Always abort on GetNumCPUs failure

### PR DESCRIPTION
**Pull request by thesamesam**
Defines a wrapper function, CheckNumCPUs, which enforces that GetNumCPUs never returns fewer than one CPU.  There is no reasonable way to continue if we are unable to identify the number of CPUs.

This is a followup to https://github.com/google/benchmark/pull/1753.

### CODE REVIEW

File: src/sysinfo.cc [hunk 1]
```diff
@@ -474,44 +474,51 @@ std::string GetSystemName() {
...
+int CheckNumCPUs(const int num_cpus) {
+  if (num_cpus < 1) {
+    PrintErrorAndDie(
+        "Unable to extract number of CPUs.  If your platform uses "
+        "/proc/cpuinfo, custom support may need to be added.");
+  }
+  return num_cpus;
+}
+
 int GetNumCPUs() {
```
Comment (LebedevRI) [inline src/sysinfo.cc:486]:
```
int GetNumCPUs() {
  int num_cpus = GetNumCPUsImpl();
  if (num_cpus < 1) {
    PrintErrorAndDie(
        "Unable to extract number of CPUs.  If your platform uses "
        "/proc/cpuinfo, custom support may need to be added.");
  }
  return num_cpus;
}

int GetNumCPUsImpl() {
```

Comment (LebedevRI) [reply src/sysinfo.cc:486]: (And place `GetNumCPUsImpl()` before `GetNumCPUs()` i guess)

File: src/sysinfo.cc [hunk 2]
```diff
@@ -474,12 +474,11 @@ std::string GetSystemName() {
...
 }
 
-int GetNumCPUs() {
+int GetNumCPUsImpl() {
 #ifdef BENCHMARK_HAS_SYSCTL
   int num_cpu = -1;
-  if (GetSysctl("hw.ncpu", &num_cpu)) return num_cpu;
-  fprintf(stderr, "Err: %s\n", strerror(errno));
-  std::exit(EXIT_FAILURE);
+  if (GetSysctl("hw.ncpu", &num_cpu)) return CheckNumCPUs(num_cpu);
```
Comment (LebedevRI) [inline src/sysinfo.cc:480]: Stale?

Comment (thesamesam) [reply src/sysinfo.cc]: Sorry!

**Review by LebedevRI — APPROVED** [review]
Cool, thank you!

**Comment by thesamesam** [comment]
Many thanks!

**Pull request merged**